frontend/src/components/tabs/InputOutputTab.tsx

```diff
@@ -0,0 +1,169 @@
+import { commonCss, padding } from 'src/Css';
+import {
+  filterEventWithInputArtifact,
+  filterEventWithOutputArtifact,
+  getLinkedArtifactsByExecution,
+  LinkedArtifact,
+} from 'src/lib/MlmdUtils';
+import { KeyValue } from 'src/lib/StaticGraphParser';
+import ArtifactPreview from '../ArtifactPreview';
+import Banner from '../Banner';
+import DetailsTable from '../DetailsTable';
+import { ExecutionTitle } from './ExecutionTitle';
+
+type ParamList = Array<KeyValue<string>>;
+
+export interface IOTabProps {
+  execution: Execution;
+  namespace: string | undefined;
+}
+
+export function InputOutputTab({ execution, namespace }: IOTabProps) {
+  const executionId = execution.getId();
+
+  // Retrieves input and output artifacts from Metadata store.
+  const { isSuccess, error, data } = useQuery<LinkedArtifact[], Error>(
+    ['execution_artifact', { id: executionId, state: execution.getLastKnownState() }],
+    () => getLinkedArtifactsByExecution(execution),
+    { staleTime: Infinity },
+  );
+
+  // Restructs artifacts and parameters for visualization.
+  const inputParams = extractInputFromExecution(execution);
+  const outputParams = extractOutputFromExecution(execution);
+  let inputArtifacts: ParamList = [];
+  let outputArtifacts: ParamList = [];
+  if (isSuccess && data) {
+    inputArtifacts = getArtifactParamList(filterEventWithInputArtifact(data));
+    outputArtifacts = getArtifactParamList(filterEventWithOutputArtifact(data));
+  }
+
+  return (
+    <ErrorBoundary>
+      <div className={commonCss.page}>
+        <div className={padding(20)}>
+          <ExecutionTitle execution={execution} />
+
+          {error && (
+            <Banner
+              message='Error in retrieving Artifacts.'
+              mode='error'
+              additionalInfo={error.message}
+            />
+          )}
+
+          <div className={commonCss.header}>{'Input'}</div>
+          <DetailsTable
+            key={`input-parameters-${executionId}`}
+            title='Parameters'
+            fields={inputParams}
+          />
+
+          <DetailsTable<string>
+            key={`input-artifacts-${executionId}`}
+            title='Artifacts'
+            fields={inputArtifacts}
+            valueComponent={ArtifactPreview}
+            valueComponentProps={{
+              namespace: namespace,
+            }}
+          />
+
+          <div className={padding(20, 'tb')} />
+          <hr />
+
+          <div className={commonCss.header}>{'Output'}</div>
+          <DetailsTable
+            key={`output-parameters-${executionId}`}
+            title='Parameters'
+            fields={outputParams}
+          />
+
+          <DetailsTable<string>
+            key={`output-artifacts-${executionId}`}
+            title='Artifacts'
+            fields={outputArtifacts}
+            valueComponent={ArtifactPreview}
+            valueComponentProps={{
+              namespace: namespace,
+            }}
```

Comment:
To fully replace ML Metadata tab content, the artifact names in the list should be links to MLMD artifact details page.

Reply:
Sounds good, I have added a logic to provide JSX.Element for artifactName to direct link to artifact details.